src/lib/split-view/split-view-panel/split-view-panel.ts

```diff
@@ -0,0 +1,214 @@
+import { CustomElement, attachShadowTemplate, FoundationProperty, coerceBoolean, coerceNumber } from '@tylertech/forge-core';
+import { tylIconDragVerticalVariant } from '@tylertech/tyler-icons/extended';
+import { tylIconDragHandle } from '@tylertech/tyler-icons/standard';
+
+import { BaseComponent, IBaseComponent } from '../../core/base/base-component';
+import { SplitViewPanelPosition, SPLIT_VIEW_PANEL_CONSTANTS } from './split-view-panel-constants';
+import { SplitViewPanelFoundation } from './split-view-panel-foundation';
+import { SplitViewPanelAdapter } from './split-view-panel-adapter';
+import { SplitViewOrientation } from '../split-view/split-view-constants';
+import { ISplitViewBase } from '../core/split-view-base';
+import { IconComponent, IconRegistry } from '../../icon';
+import { RippleComponent } from '../../ripple';
+
+import template from './split-view-panel.html';
+import styles from './split-view-panel.scss';
+
+export interface ISplitViewPanelComponent extends ISplitViewBase, IBaseComponent {
+  position: SplitViewPanelPosition;
+  size: number | string;
+  min: number;
+  max: number | undefined;
```

Comment:
Since `size` accepts a percentage value, I think both `min` and `max` properties should also accept the same. Maybe we could also allow for the `px` to be specified and just parse it out? I feel as though people may provide values like `300px` or just `300` as well things like `50%`.

Which also brings me to another though, can we enable the usage of CSS custom properties for these things. It feels like a stylistic value like this should be set via CSS (although I know we calculate everything via JavaScript). Sam and I spoke about this, and it's doable, but maybe a future feature. Just wanted to note it here for others to comment.

Reply:
I've been thinking about this a lot. Setting size constraints through custom properties is a big ask — we'll need to be able to keep track of the min and max sizes to make the accessibility features work and I'm just not seeing a way to do that. Setting the initial size via CSS works fine, but it begins to break down when I think about what happens when the user resizes.

I'd really like to offload as much stylistic and mathematical work to CSS as possible but it'll take time to figure out how to do that without compromising anything.